internal/loader/configuration_setting_loader.go

```diff
@@ -50,7 +50,7 @@ type TargetKeyValueSettings struct {
 
 type TargetSecretReference struct {
 	Type                  corev1.SecretType
-	UriSegments           map[string]KeyVaultSecretUriSegment
+	UriSegments           map[string]KeyVaultSecretMetadata
```

Comment:
The name `UriSegments` does not make sense now, how about updated to `SecretMetadatas`